test/hotspot/jtreg/runtime/Thread/TestAlwaysPreTouchStacks.java

```diff
@@ -82,6 +72,76 @@ static private final Thread createTestThread(int num) {
         t.setDaemon(true);
         return t;
     }
+    private static class ReservedCommitted {
+     public long reserved, committed;
+     public ReservedCommitted(long r, long c) {
+      reserved = r;
+      committed = c;
+     }
+    }
+
+    private static ReservedCommitted runPreTouchTest(boolean preTouch) throws Exception {
+      long reserved = 0L, committed = 0L;
+      ArrayList<String> vmArgs = new ArrayList<>();
+      Collections.addAll(vmArgs,
+              "-XX:+UnlockDiagnosticVMOptions",
+              "-Xmx100M",
+              "-XX:NativeMemoryTracking=summary", "-XX:+PrintNMTStatistics");
+      if (preTouch){
+          vmArgs.add("-XX:+AlwaysPreTouchStacks");
+      }
+      if (System.getProperty("os.name").contains("Linux")) {
+          vmArgs.add("-XX:-UseMadvPopulateWrite");
+      }
+      Collections.addAll(vmArgs, "TestAlwaysPreTouchStacks", "test");
+      ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(vmArgs);
+      OutputAnalyzer output = new OutputAnalyzer(pb.start());
+      output.reportDiagnosticSummary();
```

Comment:
Pre-existing, but this should go at the end, otherwise if a test fails you will get two summaries printed.

Reply:
If any of the checks fail they will automatically call `reportDiagnosticSuummary` so you only need to call it directly if you want the output in the log when the test passes. Hence it should go at the end.